dev/com.ibm.ws.dynacache/src/com/ibm/ws/cache/CacheConfig.java

```diff
@@ -455,6 +459,8 @@ public CacheConfig(Properties properties, CacheConfig config) {
         // -------------------------------------------------
         FieldInitializer.initFromSystemProperties(this);
         overrideCacheConfig(properties);
+        _passedInProperties.putAll(properties);
+        _passedInProperties.putAll(System.getProperties());
```

Comment:
If the values of the system properties are changed by another thread while calling putAll it will lead to a ConcurrentModificationException:

>Stack Dump = java.util.ConcurrentModificationException
>at java.util.Hashtable$Enumerator.next(Hashtable.java:1506)
>at java.util.concurrent.ConcurrentHashMap.putAll(ConcurrentHashMap.java:1094)
>at com.ibm.ws.cache.CacheConfig.<init>(CacheConfig.java:382)
>at com.ibm.ws.cache.CacheServiceImpl.parsePropertiesFromOSGiConfigAdmin(CacheServiceImpl.java:210)
>at com.ibm.ws.cache.CacheServiceImpl.start(CacheServiceImpl.java:140)